behaviors/behaviors.py

```diff
@@ -98,6 +100,40 @@ def released(self):
         return self.release_date and self.release_date < timezone.now()
 
 
+class Slugged(models.Model):
+    """
+    An abstract behavior representing adding a unique slug to a model
+    based on the slug_source property.
+    """
+    slug = models.SlugField(max_length=255, unique=True)
+
+    class Meta:
+        abstract = True
+
+    def save(self, *args, **kwargs):
+        if not self.slug:
```

Comment:
We need to add a test case that covers the `True` condition of this, basically that the slug doesn't change when you re-save an object to not lose coverage